server/api/api.go

```diff
@@ -0,0 +1,129 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-plugin-autolink/server/autolink"
+)
+
+const PluginIDContextValue = "pluginid"
```

Comment:
See my comment in the WF PR - maybe use the header for it?

Reply:
It feels strange to use a header for it. Mutating the request after it has been received just seems strange. Maybe the MM server should include it for reference.

Reply:
If this code were a proxy, it'd be adding/mutating headers - why is it strange? We add the UserID this way. But I don't think it's all that important.